### PR TITLE
Feature: raw clipboard access (WIP)

### PR DESCRIPTION
**Pull request by livecodefraser**

### CODE REVIEW

File: engine/src/exec-pasteboard.cpp [hunk 1]
```diff
@@ -290,6 +297,36 @@ void MCPasteboardEvalClipboardKeys(MCExecContext& ctxt, MCStringRef& r_string)
...
+{
+    // Get a reference to the system clipboard
+    MCRawClipboard* t_clipboard = MCRawClipboard::getSystemClipboard();
+    
+    // TODO: support multiple items
+    // Get the first item on the clipboard
+    MCAutoRefcounted<const MCRawClipboardItem> t_item = t_clipboard->getItemAtIndex(0);
+    if (t_item == NULL)
+    {
+        r_string = kMCEmptyString;
```
Comment (runrevmark) [inline engine/src/exec-pasteboard.cpp:461]: You can do `r_string = MCValueRetain(kMCEmptyString)` here

**Comment by livecodestephen** [comment]
As `MCMixinRefcounted` is a generic template wouldn't it make sense to put it in a lower level library and not in engine?

**Comment by runrevmark** [comment]
@livecodestephen: It could always be moved at a later date... The only 'lower-level' library we have at the moment is libfoundation and it doesn't quite sit there since the purpose of that library is MCValueRef types and related methods.

**Comment by runrevmark** [comment]
This looks great - the only thing is a style one... In general, the C++ style I at least have always tried to adhere to is TitleCase for everything. So - GetItemAtIndex and MCAutoRefCounted.

**Comment by livecodefraser** [comment]
I've been writing too much JavaScript-styled code recently ;)

**Comment by livecodefraser** [comment]
Closing as this is now targeted for 8.0.

**Pull request closed without merging**